### PR TITLE
Add CI and compat shims for Julia 1.3

### PR DESCRIPTION
**Pull request by staticfloat**
This should allow us to use Tar.jl within 1.3, and will pave the way for CI goodness.

### CODE REVIEW

**Comment by staticfloat** [comment]
@StefanKarpinski it looks like perhaps the windows tests are failing because we have overrun a windows file path limit?  Could that be the case?

File: src/utils.jl [hunk 1]
```diff
@@ -25,3 +25,11 @@ end
...
     Dict(f => field_data(buf, f) for f in keys(FIELDS))
 end
+
+@static if VERSION <= v"1.3"
+    unsorted_readdir(args...) = readdir(args...)
+else
+    unsorted_readdir(args...) = readdir(args...; sort=false)
+end
+
+
```
Comment (StefanKarpinski) [inline src/utils.jl]: Seems like a lot of blank lines...

File: src/Tar.jl [hunk 2]
```diff
@@ -11,6 +11,7 @@ end
 include("header.jl")
 include("create.jl")
 include("extract.jl")
+include("utils.jl")
```
Comment (StefanKarpinski) [inline src/Tar.jl]: Do we really want to load the utils file just for the one function?

**Comment by StefanKarpinski** [comment]
The 260 path limit is sooo short. How does anything work? Can we turn that off somehow? It seems possible to disable the 260 char limit from a bit of googling. Otherwise we're not testing very much on Windows.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  - linux
+  - osx
+  - windows
+julia:
+  - 1.3
+  - nightly
+notifications:
+  email: false
+after_success:
+  - julia -e 'using Pkg; Pkg.add("Coverage"); using Coverage; Coveralls.submit(Coveralls.process_folder())';
```
Comment (visr) [inline .travis.yml]: Can simply be a `coveralls: true` at the top level as well https://docs.travis-ci.com/user/languages/julia/#coverage

File: src/Tar.jl (hunk 2 above)
Comment (staticfloat) [reply src/Tar.jl]: I figured it was an oversight that it wasn't included.  Should we delete the other stuff in there, or is it useful to have around but not `include()`ed?

**Comment by staticfloat** [comment]
On a local windows machine I attempted to [use `\\?\` style paths](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#maximum-path-length-limitation) as well as setting [`LongPathsEnabled`](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#enable-long-paths-in-windows-10-version-1607-and-later) within the registry and group policy, both to no avail.  I'm not sure if this is because of interactions between libuv and windows, or if we would need to include a win32 manifest alongside Julia to make this work or whatnot, but it seems difficult to get working properly, so I'm giving up for now and I'm going to try just filtering overall path lengths.

File: test/runtests.jl [hunk 4]
```diff
@@ -12,11 +12,20 @@ end
 function make_test_tarball()
     root = mktempdir()
     let i = 0, paths = String[]
-        for a in [0, 10, 154, 155, 156, 255]
+        a_lengths = [0, 10, 154, 155, 156, 255]
+        @static if Sys.iswindows()
+            filter!( a -> 2 + a + length(root) < 260, a_lengths)
```
Comment (StefanKarpinski) [inline test/runtests.jl]:
```suggestion
            filter!(a -> 2 + a + length(root) < 260, a_lengths)
```

File: src/Tar.jl (hunk 2 above)
Comment (StefanKarpinski) [reply src/Tar.jl]: It's useful for debugging: it lets you read a 512-byte header block and then parse out the individual fields.

**Pull request merged**